compositor_pipeline/src/queue/internal_queue.rs

```diff
@@ -36,6 +38,7 @@ impl InternalQueue {
 
     pub fn enqueue_frame(&mut self, input_id: InputId, mut frame: Frame) -> Result<(), QueueError> {
         let next_output_buffer_pts = self.get_next_output_buffer_pts();
+        self.call_input_listeners(&input_id);
```

Comment:
Frames queueing from the user perspective is an implementation detail. 
From the user's perspective, it's more important when frames from certain streams are being composed, rather than enqueued. 
With current queueing logic, input frames will always be used in the next `frame set`,  meaning this implementation has the same effect, as checking the output of the queue, but relaying on that seems weird to me. What do you think?

Reply:
I don't think there is anything wrong with this approach if it gives the correct result. It might make a difference if we want to make sure that the next frame is what will be used for next render, but supporting that would be a lot harder (it would not be enough to just move that call to different location)